### ISSUE TITLE
autoPreventDefault and allowing the canvas to pan/zoom/scroll the page

### ISSUE THREAD

**Issue by Username-is-already-taken** [pixijs/pixi.js#2483]
By default, the Pixi canvas/display-area cannot be used to scroll the webpage that contains it. Which is important on touch screens. (eg. If you use the rest of the web-page to pinch-zoom into the Pixi canvas, you can become trapped and unable to zoom back out (or pan away), because there's no non-Pixi-canvas area of the page to "grab" with your pinch gesture).

To enable this functionality, I use `autoPreventDefault`.
But this comes with some undesirable side-effects, like scroll/pinch-zoom actions over the canvas registering "taps" or clicks in a way that doesn't make sense. (ie. I'm attempting to zoom or scroll the outer page at that point, not interact with the Pixi canvas)

To work around that, I modify and compile my own custom version of Pixi where I can apply preventDefault in a more granular way...

To get page-scrolling functionality it seems I only need to preventDefault in the `InteractionManager.prototype.onTouchEnd` function. Whereas `autoPreventDefault` will also preventDefault on 3 other events. (onMouseDown, onTouchMove, onTouchStart).

Leaving `autoPreventDefault = false` and applying preventDefault only to `onTouchEnd`, gives me the functionality I need. But it'd be nice to not have to customize and rebuild Pixi in this way every release. (Sorry if there's something else I'm missing here; I don't completely understand the event system in Pixi, or what else to do about this scroll-touch problem)

**Comment by englercj** [comment]
Is this still an issue? You should be able to turn off autoPreventDefault and then using the global Interaction events prevent default as necessary.

**Comment by Username-is-already-taken** [comment]
> Is this still an issue?

I couldn't say, the last version of Pixi.js I've used is 3.0.11

**Comment by Username-is-already-taken** [comment]
@zhuyunhe sorry for the delay, I don't log in to github very often.

The solution I found at the time was specific to the version of Pixi I mentioned, since I think they've re-worked that whole system since then. I don't know what method would be used for more recent versions of Pixi. But just in case you're using that version, here's what I did...

Edit PIXI.InteractionManager.prototype.onTouchStart
Commented out `if(PIXI.AUTO_PREVENT_DEFAULT)event.preventDefault();`
And then added to: PIXI.InteractionManager.prototype.onTouchEnd
`event.preventDefault();`

Which allowed drag touch events to (I guess) bubble-up to the browser, while still capturing others like taps. From what @englercj posted, I presume there's a cleaner way to do this now, that doesn't require editing Pixi.js itself. I haven't looked at the problem since then, so I couldn't say.
